apps/total-typescript/src/pages/books/[book]/[chapter]/index.tsx

```diff
@@ -162,6 +160,18 @@ const BookChapterRoute: React.FC<{
 
   const [isMenuOpen, setIsMenuOpen] = React.useState(false)
   const articleRef = React.useRef<HTMLDivElement>(null)
+  const addBookmarkMutation = trpc.bookmarks.addBookmark.useMutation({
+    onSuccess: (data) => {
+      toast.success('Bookmark added')
+    },
+    onError: (error) => {
+      if (error?.data?.httpStatus === 401) {
+        toast.error('Please log in to save bookmarks')
+      } else {
```

Comment:
sending useful notification if user isn't signed in instead of silently erroring out.